src/awkward/operations/ak_from_iter.py

```diff
@@ -8,18 +8,24 @@
 
 
 def from_iter(
-    iterable, highlevel=True, behavior=None, allow_record=True, initial=1024, resize=1.5
+    iterable,
+    *,
+    allow_record=True,
+    highlevel=True,
+    behavior=None,
+    initial=1024,
+    resize=1.5
 ):
```

Comment:
This is the only operation where `highlevel` and `behavior` are not the last parameters sent. Does it make sense to re-order them to have the same pattern across all operations?

Reply:
I know; I thought about that. If we had a rule like "highlevel and behavior are always last" and also a rule like "extremely internal arguments are always last," then one rule has to override the other when a function has both.

The `initial` and `resize` arguments determine how memory is allocated by `GrowableBuffer`, and I think the new `GrowableBuffer` doesn't even use `resize`. The sizes of panels in `GrowableBuffer` after the first one (which has `initial` bytes) hasn't been entirely settled. Since it's not copying into new contiguous buffers anymore, the exponential growth of new buffers is no longer needed, but making every panel the same size is probably not optimal, either, because there's probably two populations of `GrowableBuffers`: those that never exceed their first panel and those that have many panels. There should probably be two parameters: `initial` and `subsequent`, so that the first panel can be small (for the ones that don't get filled much) and subsequent panels can be large (to minimize overhead for the ones that get filled a lot), but linearly: without exponential growth.

I think it's very unlikely that any user would want to configure such details, and they might change in the future. (It will be easy to do it in a deprecation cycle: use of `resize` will raise a warning until some version, after which it gets removed.) So that's why I thought these options should be more hidden than `highlevel` and `behavior`.

But the good news is that we can now change that order at will, even after the 2.0.0 release, because users cannot depend on their order anymore.